src/Microsoft.Restier.WebApi/HttpRequestMessageExtensions.cs

```diff
@@ -40,5 +42,42 @@ public static ODataDomainChangeSetProperty GetChangeSet(this HttpRequestMessage
 
             return changeSetProperty;
         }
+
+        /// <summary>
+        /// Gets the domain factory from the <see cref="HttpRequestMessage"/>.
+        /// </summary>
+        /// <param name="request">The HTTP request.</param>
+        /// <returns>The domain factory.</returns>
+        internal static Func<IDomain> GetDomainFactory(this HttpRequestMessage request)
+        {
+            Ensure.NotNull(request, "request");
+
+            Func<IDomain> domainFactory;
+            object value;
+            if (request.Properties.TryGetValue(DomainFactoryKey, out value))
+            {
+                domainFactory = value as Func<IDomain>;
+                Contract.Assert(domainFactory != null);
+            }
+            else
+            {
+                domainFactory = null;
+            }
+
+            return domainFactory;
+        }
```

Comment:
Fixed in https://github.com/OData/RESTier/commit/2b6607439d3b5bcc0956b7ed492caa5ace04cdb5